File: app/src/main/java/org/baiyu/fucksensors/sensor/SensorEnvironment.java
package org.baiyu.fucksensors.sensor;

import android.hardware.Sensor;

public enum SensorEnvironment implements SensorType {

    TYPE_AMBIENT_TEMPERATURE(Sensor.TYPE_AMBIENT_TEMPERATURE),
    TYPE_LIGHT(Sensor.TYPE_LIGHT),
    TYPE_PRESSURE(Sensor.TYPE_PRESSURE),
    TYPE_RELATIVE_HUMIDITY(Sensor.TYPE_RELATIVE_HUMIDITY),
    // TYPE_TEMPERATURE(Sensor.TYPE_TEMPERATURE),

    ;
    private final int TYPE;

    SensorEnvironment(int type) {
        TYPE = type;
    }

    @Override
    public int getType() {
        return TYPE;
    }
}

File: app/src/main/java/org/baiyu/fucksensors/sensor/SensorEnvironment.kt
package org.baiyu.fucksensors.sensor

import android.hardware.Sensor

enum class SensorEnvironment(override val type: Int) : SensorType {
    TYPE_AMBIENT_TEMPERATURE(Sensor.TYPE_AMBIENT_TEMPERATURE),
    TYPE_LIGHT(Sensor.TYPE_LIGHT),
    TYPE_PRESSURE(Sensor.TYPE_PRESSURE),

    // TYPE_TEMPERATURE(Sensor.TYPE_TEMPERATURE),
    TYPE_RELATIVE_HUMIDITY(Sensor.TYPE_RELATIVE_HUMIDITY);
}